controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -22,9 +24,10 @@ import (
 
 // Reconciler reconciles a ToolchainCluster object
 type Reconciler struct {
-	Client     client.Client
-	Scheme     *runtime.Scheme
-	RequeAfter time.Duration
+	Client      client.Client
+	Scheme      *runtime.Scheme
+	RequeAfter  time.Duration
+	CheckHealth func(context.Context, *kubeclientset.Clientset) (bool, error)
```

Comment:
the `CheckHealth` field doesn't have to be exported - the unit test is in the same package